app/scripts/controllers/keyring-events/utils.ts

```diff
@@ -0,0 +1,54 @@
+import { has, isError } from 'lodash';
+import { isManifestV3 } from '../../../../shared/modules/mv3.utils';
+import { HARDWARE_KEYRING_INIT_OPTS } from '../../../../shared/constants/hardware-wallets';
+
+/**
+ * Returns true if text resembles MV3 error message.
+ *
+ * @param _text
+ */
+const isMv3ErrorMessage = (_text: string): boolean => {
+  const text = _text.toLowerCase();
+  const mv3ErrorText = [
+    'navigator.usb',
+    'navigator.hid',
+    'document is not defined',
+  ];
+
+  return mv3ErrorText.some((errorText) => text.includes(errorText));
+};
+
+/**
+ * Determines if error is provoked by manifest v3 changes,
+ * e.g. WebUSB incompatibility, WebHID incompatibility,
+ * and general DOM access.
+ *
+ * @param error
+ */
+export const isServiceWorkerMv3Error = (error: any): boolean => {
```

Comment:
I'm in two minds as to whether this should occur, and instead, we could just pass every error that occurs on the backend to the frontend.